tools/test/h5dump/h5dumpgentest.c

```diff
@@ -729,8 +730,16 @@ gent_softlink2(void)
     /*-----------------------------------------------------------------------
      * Named datatype
      *------------------------------------------------------------------------*/
-    datatype = H5Tcopy(H5T_NATIVE_INT);
-    status   = H5Tcommit2(fileid1, "dtype", datatype, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+    if (big_endian_committed) {
+        target_type = H5T_STD_I32BE;
+    }
+    else {
+        target_type = H5T_NATIVE_INT;
```

Comment:
It seems like it would be better here to copy the native integer type and then set its endian-ness based on the boolean parameter, as `H5T_NATIVE_INT` could be big-endian.